src/main/java/com/garbagemule/MobArena/formula/Environment.java

```diff
@@ -128,7 +128,43 @@ private void registerOperatorToken(TokenType type, String operator, List<Token>
         operators.add(i, token);
     }
 
-    @SuppressWarnings("Convert2MethodRef")
+    // math functions implemented here to get rid of "unchecked conversion" warnings...
```

Comment:
These warnings are a little hyper-aggressive, but I suppose there is an ounce of rationale to their paranoia. The actual issue is that the underlying interfaces, UnaryOperation and BinaryOperation, inherit from the _generic_ Function and BiFunction interfaces in the `java.util.function` package. Because these interfaces have generic type parameters, and because generics in Java don't support primitive types, the two operation interfaces must use `Double` instead of `double`. The warnings, then, say that e.g. `Math::sqrt(double)` requires an unchecked conversion (or _unboxing_) of the `Double` that the interface is given into the `double` that the `sqrt` function requires. This could potentially lead to null pointer exceptions, since you could technically pass a null value to a UnaryOperation object. Of course, this never actually happens, but the compiler can't know that.

There are two better ways around the warnings:

1. Absolve the UnaryOperation and BinaryOperation interfaces of their dependence on Function and BiFunction and explicitly define methods `double apply(double)` and `double apply(double, double)`, respectively, in them.
2. Switch to a lambda function like the one used for other operators, e.g. the unary +/- or the binary `^` operator.

The latter doesn't actually solve the underlying problem, because a null reference could still be passed to a UnaryOperation or a BinaryOperation. I'm not sure why VS Code doesn't continue to warn about it, but the current proposal doesn't actually solve the underlying problem either, as the new methods still take `Double` arguments instead of `double` (which they must with the current interfaces).

The former is definitely the way to go. I probably decided to use the Function and BiFunction interfaces from the standard library in an effort to try to "be correct" about it, but it doesn't actually provide any benefit whatsoever. If you're up for it, give it a shot! If not, I can work it in :)